### PR TITLE
Use of reference files in PEARL system test

### LINKED ISSUE #16459: Use reference files in PEARL system tests
This is important before refactoring the PEARL scripts.
In addition to several checks on the spectra produced by the various routines, the system tests should do comparisons against reference files. 

For a comment about this in a more specific context, with example link see: https://github.com/mantidproject/mantid/pull/16103#discussion_r62980595

Add reference files and compare results against them in both the calibration system test (#16103) and the focusing system test (#15966)

### PR DESCRIPTION
**Pull request by shahrozahmed1**
**Description of work.**

This PR enable comparisons again reference files for the `PearlPowderDiffractionScriptTest` system test, comparison being carried out for focus as well as the calibration system test. Recommend to merge this PR before merging #16590 .

**To test:**

Code Check
Check that system test build pass
No further pylint violation generated
If possible run `PearlPowderDiffractionScriptTest` system test locally

Fixes #16459 
_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Testing/SystemTests/tests/analysis/PearlPowderDiffractionScriptTest.py [hunk 1]
```diff
@@ -23,10 +23,19 @@ def requiredFiles(self):
...
                           'PEARL/Focus_Test/Calibration/pearl_offset_15_3.cal',
                           'PEARL/Focus_Test/Calibration/van_spline_TT70_cycle_15_4.nxs',
-                          'PEARL/Focus_Test/Attentuation/PRL112_DC25_10MM_FF.OUT'))
+                          'PEARL/Focus_Test/Attentuation/PRL112_DC25_10MM_FF.OUT',
+                          # reference files
+                          'PEARL/Focus_Test/RefFiles/Ref_PRL92476_92479.nxs',
+                          'PEARL/Focus_Test/RefFiles/Ref_PRL92476_92479-0.gss',
+                          'PEARL/Focus_Test/RefFiles/Ref_PRL92476_92479_d_xye-0.dat',
+                          'PEARL/Focus_Test/RefFiles/Ref_PRL92476_92479_tof_xye-0.dat'
+                          ))
```
Comment (FedeMPouzols) [inline Testing/SystemTests/tests/analysis/PearlPowderDiffractionScriptTest.py]: Good. There is a `reference` subdirectory where many system tests put their reference files in a flat structure. But I think this is better as this test has several files and also the focus part of the test and the calibration part of the test.

**Comment by FedeMPouzols** [comment]
This looks very good to me. Only after the next night builds after this is merged in we'll know if the comparison with the reference files is exact enough on platforms other than rhel7 (especially windows may misbehave). If that happens then we may need to modify the tolerance parameter. Let's cross fingers that we don't have any accuracy issues. In principle there is not so much fitting involved here so there shouldn't be big differences in the focused spectra / fitted splines across platforms.

**Comment by shahrozahmed1** [comment]
@FedeMPouzols Similar reference file comparison is also within [`PEARLSystemTest`](https://github.com/mantidproject/mantid/blob/master/Testing/SystemTests/tests/analysis/PEARLSystemTest.py#L59). The tolerance is not being set in `PEARLSystemTest` either, which has been around for a long time so it shouldn't cause any issues.

**Comment by FedeMPouzols** [comment]
This strenghtens the PEARL system tests with direct comparisons of the output workspaces/files.
Code looks good and tests passed. This will be additional insurance for refactoring and further work on PEARL reduction scripts like #16590.

:shipit:

**Pull request merged**